etcd/v1/types.go

```diff
@@ -0,0 +1,84 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterMemberRequest describes a request to join the cluster by a potential etcd cluster member.
+type ClusterMemberRequest struct {
+	// Standard object's metadata.
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// The membership request itself.
+	// +optional
+	Spec ClusterMemberRequestSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+
+	// Derived information about the request.
+	// +optional
+	Status ClusterMemberRequestStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// ClusterMemberRequestSpec defines an individual etcd cluster member request.
+type ClusterMemberRequestSpec struct {
+	// Name of the etcd peer member requesting membership.
+	Name string `json:"Name" protobuf:"bytes,1,opt,name=name"`
```

Comment:
how does this name relate to the name of the object?  Is this duplicated information?